### PR TITLE
kernel: repo-extra json with kernelrelease

### PR DESCRIPTION
**Pull request by idryomov**
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

### CODE REVIEW

**Comment by idryomov** [comment]
This puts kernelrelease string into version, since it's not necessarily a package manager version.  teuthology only cares about package_manager_version right now though - @zmc what do you think about a shaman-only builder.scmversion or so?

File: kernel/build/build_rpm [hunk 1]
```diff
@@ -84,6 +84,19 @@ if [ "$THROWAWAY" = false ] ; then
     # push binaries to chacra
     find ~/rpmbuild/ | egrep '\.rpm$' | $VENV/chacractl binary ${chacra_flags} create ${chacra_endpoint}/$ARCH/
 
+    # write json file with build info
+    cat > $WORKSPACE/repo-extra.json << EOF
+{
+    "version":"$kernelrelease",
```
Comment (andrewschoen) [inline kernel/build/build_rpm]: You'll also need a ``package_manager_version`` field as that's what teuthology will use. The value there should match the version included in the binary name.

Comment (idryomov) [inline kernel/build/build_rpm]: @andrewschoen Currently we have builder.version property that corresponds to extra:package_manager_version. @zmc and I agreed to add another property, builder.scm_version, that would correspond to extra:version, to be used by the kernel task. builder.scm_version can either be ShamanProject-specific or just return builder.version in the GitbuilderProject case.

The rationale here is that the kernel task has always been special and kernel gitbuilder /version file never contained an actual package version.

Comment (andrewschoen) [inline kernel/build/build_rpm]: @idryomov that's totally fine with me if it works for @zmc

Comment (zmc) [inline kernel/build/build_rpm]: @idryomov @andrewschoen I'm a bit confused - we do still need `package_manager_version` - but this PR doesn't contain it. If we also need `version` (and it sounds like we do), we can use that too.

Comment (idryomov) [inline kernel/build/build_rpm]: @zmc Your wip-shaman-kernel branch uses `builder.version` to assign to `need_version[role]`.  That should become `builder.scm_version`.

Unless packaging.py *requires* `package_manager_version` - which I think it doesn't - we shouldn't need `package_manager_version` for the kernel task at all.

**Pull request merged**